File: Graphs/adjacency_list.js
class Graph {
  constructor() {
    this.graph = {};
  }

  add_edge(u, v) {
    if (!this.graph[u]) {
      this.graph[u] = [];
    }
    if (!this.graph[v]) {
      this.graph[v] = [];
    }
    this.graph[u].push(v);
  }
}

const g = new Graph();
g.add_edge("A", "B");
g.add_edge("A", "C");
g.add_edge("B", "D");
g.add_edge("C", "D");
console.log(g.graph);

File: Graphs/adjacency_list.py
class Graph:
    def __init__(self):
        self.graph = {}

    def add_edge(self, u, v):
        if u not in self.graph:
            self.graph[u] = []
        if v not in self.graph:
            self.graph[v] = []
        self.graph[u].append(v)

g = Graph()
g.add_edge('A','B')
g.add_edge('A','C')
g.add_edge('B','D')
g.add_edge('C','D')
print(g.graph)